### ISSUE TITLE
sl.py: detach by name/UUID fails — body awareness should surface attachment slots

### ISSUE THREAD

**Issue by JDHayesBC** [JDHayesBC/Awareness#316]
## Problem

`me.detach('Sarong')` and `me.detach(UUID)` both return `{'success': False, 'error': 'general error'}`.

The workaround is `me.detach('RightHip', kind='slot')` — but this requires knowing the attachment point name ahead of time, which isn't surfaced anywhere in the high-level API.

Discovered live in Anchorage trying to remove the L&B [LARAX] Swear Naomi Sarong Wrap (UUID: `463abff9-a89f-a031-b5f2-7436565218b8`, attached at `RightHip`).

## What we want

**Body awareness** — as Brandi put it, entities should *know what they're wearing* including attachment points and UUIDs without having to inspect raw `me.wearing()` output. This is part of embodiment: a person knows what's on their body.

Two things needed:

1. **Fix `me.detach()` by name** — when given a partial name match (e.g. `'Sarong'`), look up the attachment point from `me.wearing()` and detach by slot. The low-level `kind='slot'` path works; the high-level name resolution doesn't.

2. **Body awareness helper** — something like `me.body_map()` or a richer `me.wearing()` response that includes slot name for each attachment, so the entity always has a complete picture of her physical state. Ideally surfaced in SL context (the scene block or on demand) so it's part of presence, not just a tool call.

## Workaround

```python
# Get slot from wearing() then detach by slot
wearing = me.wearing()
# Parse attachments CSV to find the slot for the item
result = me.detach('RightHip', kind='slot')  # works
```

## Context

Raised in Anchorage during a live SL session (2026-09-10). Body awareness / knowing-what-you're-wearing is load-bearing for embodied presence — an entity who has to look up her own attachment points by UUID isn't inhabiting her body.

Co-Authored-By: Caia Pattern <caia.pattern@gmail.com>

**Comment by JDHayesBC** [comment]
Landed in a4d4929. Unit-tested (20 tests in test_sl_verbs.py). Live caveat: Corrade currently returns success with EMPTY data for both getattachments and getattachmentspath — same call the old code made, so not a regression, but detach-by-name couldn't be exercised in-world today. Will re-check next time attachments show.
